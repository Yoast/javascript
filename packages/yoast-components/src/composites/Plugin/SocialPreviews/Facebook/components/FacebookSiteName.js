import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";

const FacebookSiteNameWrapper = styled.p`
	color: #606770;
	font-size: 12px;
	line-height: 11px;
	text-transform: uppercase;
	overflow: hidden;
	text-overflow: ellipsis;
	white-space: nowrap;
`;

/**
 * Renders a FacebookSiteName component.
 *
 * @param {object} props The props.
 *
 * @returns {React.Element} The rendered element.
 */
const FacebookSiteName = ( props ) => {
	return (
		<FacebookSiteNameWrapper>
			{ props.siteName }
		</FacebookSiteNameWrapper>
	);
};

FacebookSiteName.propTypes = {
	siteName: PropTypes.string.isRequired,
};

export default FacebookSiteName;
