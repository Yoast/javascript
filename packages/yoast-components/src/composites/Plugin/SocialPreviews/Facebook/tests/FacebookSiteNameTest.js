/* External dependencies */
import React from "react";
import renderer from "react-test-renderer";

/* Internal dependencies */
import FacebookSiteName from "../components/FacebookSiteName";

describe( "FacebookSiteName", () => {
	it( "matches the snapshot by default", () => {
		const component = renderer.create(
			<FacebookSiteName siteName="sitename.com" />
		);

		const tree = component.toJSON();
		expect( tree ).toMatchSnapshot();
	} );
} );
