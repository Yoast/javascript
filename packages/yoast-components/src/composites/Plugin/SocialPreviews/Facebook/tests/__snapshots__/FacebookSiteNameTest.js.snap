// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FacebookSiteName matches the snapshot by default 1`] = `
.c0 {
  color: #606770;
  font-size: 12px;
  line-height: 11px;
  text-transform: uppercase;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

<p
  className="c0"
>
  sitename.com
</p>
`;
